index.js

```diff
@@ -2,5 +2,35 @@
 
 var window = require('global/window')
 
+var OfflineContext = window.OfflineAudioContext || window.webkitOfflineAudioContext
 var Context = window.AudioContext || window.webkitAudioContext
-if (Context) module.exports = new Context()
+
+var cache = {}
+
+module.exports = function getContext (options) {
+	if (!Context) return null
+
+	if (typeof options === 'number') {
+		options = {sampleRate: options}
+	}
+
+	var sampleRate = options && options.sampleRate
+
+
+	if (options && options.offline) {
+		if (!OfflineContext) return null
+
+		return new OfflineContext(options.channels || 2, options.length, sampleRate || 44100)
+	}
+
+
+	//cache by sampleRate, rather strong guess
+	var ctx = cache[sampleRate]
```

Comment:
Is there a case where you'd have the same sample rate, but other options that are different?  Wouldn't that make it ignore the other options and give you a different context?

That is why I went with the single context cache approach in mine, but wasn't completely sure.